OpenRA.Mods.Common/Traits/PaletteEffects/RotationPaletteEffect.cs

```diff
@@ -92,8 +92,8 @@ public void AdjustPalette(IReadOnlyDictionary<string, MutablePalette> palettes)
 
 			foreach (var kvp in palettes)
 			{
-				if ((info.Palettes.Count > 0 && !AnyPaletteNameStartsWith(info.Palettes, kvp.Key))
-					|| (info.ExcludePalettes.Count > 0 && AnyPaletteNameStartsWith(info.ExcludePalettes, kvp.Key)))
+				if ((info.Palettes.Count > 0 && !StartsWithAny(kvp.Key, info.Palettes))
+					|| (info.ExcludePalettes.Count > 0 && StartsWithAny(kvp.Key, info.Palettes)))
```

Comment:
Typo: `info.ExcludePalettes`.  Right now you're checking against `info.Palettes` again.